### PR TITLE
Update smarter encryption to latest bloom and remove service

### PR DESCRIPTION
**Pull request by subsymbolic**
Task/Issue URL: https://app.asana.com/0/361428290920652/1195088920676364/f

**Description**:
- Update to latest bloom filter
- Embed smarter encryption data in app
- Remove backend calls to smarter encryption

**Steps to test this PR**:

Test fresh installation with backend data:
1.  Install a fresh installation of the app, wait for all data to sync
1. Navigate to http://facebook.com and ensure this is upgraded
1. Navigate to some long url that doesn't exist and is thus definitely not in our list e.g http://thisUrlDoesNotExistAndIsNotInOurSmarterEncryptionList. com and ensure this is not upgraded

Test fresh installation with embedded data:
1. Block all calls to out smarter encryption endpoints in Charles these are https://staticcdn.duckduckgo.com/https/https* or make breaking changes to the urls at AppUrls.swift so they definitely fail
1.  Install a fresh installation of the app
1. Navigate to http://facebook.com and ensure this is upgraded
1. Navigate to some long url that doesn't exist and is thus definitely not in our list e.g http://thisUrlDoesNotExistAndIsNotInOurSmarterEncryptionList. com and ensure this is not upgraded

Test migration:
1. Install and run app from develop, making sure that all data is downloaded
1. Now install this branch over the top
1. Ensure no db migration crashes occur
1. Navigate to http://facebook.com and ensure this is upgraded
1. Navigate to some long url that doesn't exist and is thus definitely not in our list e.g http://thisUrlDoesNotExistAndIsNotInOurSmarterEncryptionList. com and ensure this is not upgraded

###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Review by cmonfortep — COMMENTED** [review]
Everything looks good, just a few minor comments.

Good job here, we did a pretty nice refactor.

File: app/src/main/java/com/duckduckgo/app/httpsupgrade/HttpsUpgrader.kt
Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/httpsupgrade/HttpsUpgrader.kt:103]: I could lock around `bloomFilter?.contains(host)` check but I don't really want to make everyone wait for reads, just for writes to complete, especially on startup when we don't have a bloom filter yet. In the unlikely event that something grabbed the lock just are we released it, we would just read from the current bloomFilter variable rather that the one about to be written, which is acceptable. Happy to zoom if you want to chat about this more.

Comment (cmonfortep) [reply app/src/main/java/com/duckduckgo/app/httpsupgrade/HttpsUpgrader.kt:103]: Thanks. Sounds good not to block each read. 👍

File: app/src/main/java/com/duckduckgo/app/httpsupgrade/api/HttpsUpgradeDataDownloader.kt
Comment (subsymbolic) [reply app/src/main/java/com/duckduckgo/app/httpsupgrade/api/HttpsUpgradeDataDownloader.kt:30]: As it is preexisting logic beyond the scope of this PR I will defer it to get the release out. I think that's great feedback though and I would under normal circumstances include it! If it's any consolation, we extracted a bunch of logic from this class which wasn't tested and is now covered by the integration tests 😁

**Review by cmonfortep — APPROVED** [review]
LGTM Thanks.

BTW, bitrise is currently failing, but the work done here is unrelated to that. Feel free to merge this PR even if bitrise fails.

**Comment by marcosholgado** [comment]
> LGTM Thanks.
> 
> BTW, bitrise is currently failing, but the work done here is unrelated to that. Feel free to merge this PR even if bitrise fails.

Just make sure the tests pass locally before merging 👍

**Comment by subsymbolic** [comment]
Thanks for the review!

**Comment by subsymbolic** [comment]
Merging as per previous comments despite BitRise tests issues across our branches. Tests passing fine locally and on one BitRise (Nexus 5) device.
tests_passing_on_Nexus5

**Pull request merged**
